networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -1935,3 +1935,88 @@ def dist_path(v):
 
     return {v: dist_path(v) for v in G}
 
+def ctp(G, blocked, starting, ending, weight='weight'):
+    """
+    Canadian Travelling Problem (CTP) aimed to find the best feasible path given
+    some blocked edges.
+
+    CTP solved using a repositioning algorithm.
+    First, the initial path is found using Dijkstra's algorithm and then the
+    repostitioning algorithm is implemented using taking into consideration
+    the edges that are blocked.
+
+    Parameters
+    ----------
+    G: A weighted graph.
+
+    blocked: list
+        A list with the blocked edges.
+
+    starting: node
+        Starting node.
+
+    ending: node
+        Ending node.
+
+    weight: string, optional (default='weight')
+        Represents the edge weight of each edge data key.
+
+    Returns
+    -------
+    dis: list
+        The path followed taking into consideration the blocked edges.
+
+    Example
+    -------
+    >>> import networkx as nx
+    >>> e = [(0, 1, 4), (0, 2, 6), (1, 2, 1), (1, 3, 8), (3, 2, 1), (3, 4, 16),
+    ...        (4, 2, 1), (4, 5, 32), (5, 2, 1)]
+    >>> G = nx.MultiGraph()
+    >>> G.add_weighted_edges_from(e)
+    >>> blocked = [(1,2), (3,2), (4,2)]
+    >>> output = ctp(G, blocked, 0, 2)
+    >>> print(output)
+    [0, 2]
+
+    Notes
```

Comment:
It's very important to note that this problem is PSPACE-complete and that this (exact) algorithm does not run in polynomial time.